packages/svelte/src/internal/client/proxy.js

```diff
@@ -70,16 +64,17 @@ export function proxy(value, immutable = true, owners) {
 			});
 
 			if (DEV) {
-				// set ownership — either of the parent proxy's owners (if provided) or,
-				// when calling `$.proxy(...)`, to the current component if such there be
 				// @ts-expect-error
-				value[STATE_SYMBOL].o =
-					owners === undefined
-						? current_component_context
+				value[STATE_SYMBOL].parent = parent;
+
+				// @ts-expect-error
+				value[STATE_SYMBOL].owners =
+					parent === null
+						? current_component_context !== null
 							? // @ts-expect-error
 								new Set([current_component_context.function])
 							: null
-						: owners && new Set(owners);
+						: parent.owners;
```

Comment:
This (084aabf5a09abb02b031e05f7b23a8bce955ae98) defeats the object. We can't just share an `owners` set between different objects, because then if you have something like this...

```svelte
<script>
  import Counter from './Counter.svelte';

  let object = $state({
    shared: { count: 0 },
    notshared: { count: 0 }
  });
</script>

<Counter
  bind:shared={object.shared}
  notshared={object.notshared}
/>
```

...adding `<Counter>` as an owner of `shared` will also add it as an owner of `notshared`.

Reply:
Is that really necessary to differentiate? Feels like a pretty rare case, and it would only be a false negative, not a false positive.
If you want to keep the behavior, then we need a separate special "ownerless" value, regardless of my change to widen ownership.